crates/runtime/src/modelruntime.rs

```diff
@@ -0,0 +1,22 @@
+use arrow::record_batch::RecordBatch;
+use snafu::prelude::*;
+
+pub mod tract;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    TractError { source: tract_core::anyhow::Error },
+
+    ArrowError { source: arrow::error::ArrowError },
+
+    ShapeError { source: ndarray::ShapeError },
+}
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub trait Runnable: Send + Sync {
+    fn run(&self, input: Vec<RecordBatch>) -> Result<RecordBatch>;
```

Comment:
For traits, I've come to the conclusion that it doesn't make sense to define concrete error types - as you then need to leak implementation details about which error can occur in the trait definition. It's better to just return `std::result::Result<RecordBatch, Box<dyx std:error::Error>>`

Reply:
that makes more sense. I prefer what you mentioned, working on the change

Reply:
To be clear, the actual implementations should still use Snafu - but the trait itself doesn't need it.